### ISSUE TITLE
v0.9.17 LockCachingAudioSource load cannot get duration for some audio

### ISSUE THREAD

**Issue by akindone** [ryanheise/just_audio#589]
**Which API doesn't behave as documented, and how does it misbehave?**
v0.9.17 cannot get the duration for some audio; but v0.9.16 works fine

**Minimal reproduction project**

"The example".
replace the example_caching.dart url to 'https://focus-resource.oss-cn-beijing.aliyuncs.com/zen/course/mp3/all/32eb79105112420e855cebc8b860bc60.mp3'

**To Reproduce (i.e. user steps, not code)**
1. See error that: the duration is always 00:00 and I print the playbackEventStream event it status has been ready

**Logs**

```
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.idle, updateTime=2021-11-12 15:11:57.262344, updatePosition=0:00:00.000000, bufferedPosition=0:00:00.000000, duration=null, currentIndex=null}
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.idle, updateTime=2021-11-12 15:11:57.945072, updatePosition=0:00:00.000000, bufferedPosition=0:00:00.000000, duration=null, currentIndex=0}
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.loading, updateTime=2021-11-12 15:11:57.945072, updatePosition=0:00:00.000000, bufferedPosition=0:00:00.000000, duration=null, currentIndex=0}
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.loading, updateTime=2021-11-12 15:11:58.064, updatePosition=0:00:00.000000, bufferedPosition=0:00:00.000000, duration=null, currentIndex=0}
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.ready, updateTime=2021-11-12 15:11:58.064, updatePosition=0:00:00.000000, bufferedPosition=0:00:05.520000, duration=null, currentIndex=0}
I/flutter (24964): playbackEventStream event: {processingState=ProcessingState.ready, updateTime=2021-11-12 15:11:58.064, updatePosition=0:00:00.000000, bufferedPosition=0:01:27.072000, duration=null, currentIndex=0}
```

**Expected behavior**
get the correct duration

**Screenshots**
![image](https://user-images.githubusercontent.com/17843738/141427507-29fb96fa-6e22-416b-8879-1dcd65277af7.png)

**Smartphone (please complete the following information):**
 - Device: vivo 
 - OS: Android 

**Flutter SDK version**
```
Flutter (Channel stable, 2.5.1, on macOS 11.5.2 20G95 darwin-x64, locale zh-Hans-CN)
```

**Comment by ryanheise** [comment]
I can reproduce it. Interesting... I suspect there's a difference in the headers being output by the proxy server since I did try to make it more closely follow the range request spec.

**Comment by ryanheise** [comment]
Logging the response headers, first on the previous version:

```
I/flutter (22171): ### request null-null
I/flutter (22171): ### content-type: audio/mpeg
I/flutter (22171): ### accept-ranges: bytes
I/flutter (22171): ### Content-Length: 28928763
I/flutter (22171): ### Status: 200
I/flutter (22171): ### loaded duration: 0:20:05.045000
```

Current version:

```
I/flutter (22171): ### request null-null
I/flutter (22171): ### content-type: audio/mpeg
I/flutter (22171): ### accept-ranges: bytes
I/flutter (22171): ### Content-Length: -1
I/flutter (22171): ### Status: 200
I/flutter (22171): ### loaded duration: null
```

So the content length is not being set on 200 responses. I'll push a fix shortly.

**Comment by MrRoboots** [comment]
merge pull requests  HLS Headers Now Working
It can be solved

**Comment by ryanheise** [comment]
@MrRoboots I think you may be commenting on the wrong issue. I'll mark both of our comments as off topic.

**Comment by ryanheise** [comment]
I've just pushed a fix in `fix/stream_content_length`. Can you let me know how it goes?

**Comment by caldart** [comment]
I'm having the same issue.
Just as curiosity, when I play the file and replay using player.seek(Duration.zero) so the duration is displayed normally.

**Comment by ryanheise** [comment]
> I'm having the same issue.

Hmm that's surprising, I thought I just fixed it as per the comment I made just before yours. Can you share which audio URL you're using that causes this fix to fail?

**Comment by caldart** [comment]
Sure. 
https://portal.homeenglish.com.br/uploads-aula/C_IM_S_A_03_groceries.mp3

**Comment by ryanheise** [comment]
And which device did you test the above fix on?

**Comment by caldart** [comment]
Im using emulator on Windows (Pixel 2 API 24)

**Comment by ryanheise** [comment]
You're saying you're using an emulator but I'm asking you to confirm that you are indeed testing the "fix" that I keep mentioning above, not just what URL and what device but are you actually testing the fix. I notice each reply you make doesn't acknowledge my comment where I fixed it so I'm not sure if you are actually testing the fix. Are you testing the fix that I mentioned above?

**Comment by caldart** [comment]
I'm sorry dude for bothering you, I'm kind noob and I was doing the correction but made some mess.
Anyway, I get to fix now and the fix you posted it is working perfectly.
Tks

**Comment by ryanheise** [comment]
@caldart glad to know that the fix is working for you.

@akindone is the `fix/stream_content_length` branch working for you?

**Comment by akindone** [comment]
@ryanheise the `fix/stream_content_length` branch working for me, thanks.

**Comment by ryanheise** [comment]
Thanks @caldart and @akindone both for testing :+1:

I will publish the fix at my next opportunity.

**Comment by ryanheise** [comment]
The fix is now published in 0.9.18.
